spec/services/create_next_edition_service_spec.rb

```diff
@@ -0,0 +1,121 @@
+
+      expect(next_edition.draft?).to be(true)
+    end
+
+    it "resets change_note, update_type and propoposed_publish_time values" do
+      current_edition = create(:edition,
+                               :published,
+                               change_note: "note",
+                               update_type: "minor",
+                               proposed_publish_time: Time.zone.now)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_note).to be_empty
+      expect(next_edition.update_type).to eq("major")
+      expect(next_edition.proposed_publish_time).to eq(nil)
+    end
+
+    it "appends the change note details when there is a change note and a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "major")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history.first).to match a_hash_including(
+        "note" => "note",
+        "public_timestamp" => Date.yesterday.noon.rfc3339,
+      )
+    end
+
+    it "does not update the change history when the current edition is not a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "minor")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to be_empty
+    end
+
+    it "does not update the change history when the current edition lacks a change note" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               change_note: "")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to be_empty
+    end
+
+    context "when the current edition is the first edition" do
+      it "doesn't append the change note details" do
+        current_edition = create(:edition,
+                                 :published,
+                                 change_note: "note")
+
+        next_edition = described_class.call(current_edition: current_edition,
+                                            user: user)
+
+        expect(next_edition.change_history).to be_empty
+      end
```

Comment:
Similar points about the eq change_history on this and above test

Reply:
See previous comment.